reviews/serializers.py

```diff
@@ -2,53 +2,76 @@
 from __future__ import unicode_literals
 
 from rest_framework import serializers as ser
+from rest_framework.fields import SkipField
+from transitions import MachineError
 
 from api.base.utils import absolute_reverse
+from api.base.utils import get_user_auth
 from api.base.serializers import JSONAPISerializer
 from api.base.serializers import RelationshipField
 from api.base.serializers import LinksField
 
+from reviews.workflow import Actions
 from reviews.models import ReviewLog
-from reviews.models import ReviewProviderMixin
-from reviews.models import ReviewableMixin
+from reviews.permissions import has_reviews_permission
+
+
+# Pseudo-class to hide the creator field if it shouldn't be shown
+def IfNotAnonymous(field_cls):
+    class Field(field_cls):
+        def get_attribute(self, instance):
+            request = self.context.get('request')
+            view = self.context.get('view')
+            auth = get_user_auth(request)
+            if view and request and auth.logged_in:
+                comments_anonymous = instance.reviewable.provider.reviews_comments_anonymous
+                if not comments_anonymous or has_reviews_permission(auth.user, instance, 'view_logs'):
+                    return super(Field, self).get_attribute(instance)
+            raise SkipField
+    return Field
 
 
 class ReviewLogSerializer(JSONAPISerializer):
     filterable_fields = frozenset([
         'id',
+        'action',
         'from_state',
         'to_state',
         'date_created',
         'date_modified',
-        'creator',
         'provider',
         'reviewable',
     ])
 
-    id = ser.CharField(source='_id', required=True)
+    id = ser.CharField(source='_id', read_only=True)
 
-    from_state = ser.CharField(max_length=15)
-    to_state = ser.CharField(max_length=15)
+    action = ser.CharField(max_length=31)
 
     # TODO what limit do we want?
-    comment = ser.CharField(max_length=65535)
+    comment = ser.CharField(max_length=65535, required=False)
+
+    from_state = ser.CharField(max_length=31, read_only=True)
+    to_state = ser.CharField(max_length=31, read_only=True)
 
     date_created = ser.DateTimeField(read_only=True)
     date_modified = ser.DateTimeField(read_only=True)
 
     provider = RelationshipField(
         related_view='preprint_providers:preprint_provider-detail',
         related_view_kwargs={'provider_id': '<reviewable.provider._id>'},
+        filter_key='reviewable__provider___id',
     )
 
     reviewable = RelationshipField(
```

Comment:
Random thought, what if we just made reviewables a FK to the GUID of the object?
Literally just a spitball, let's not change anything now.